utils/python/CIME/test_scheduler.py

```diff
@@ -148,13 +148,17 @@ def __init__(self, test_names, test_data=None,
 
         # Setup phases
         self._phases = list(PHASES)
+        # These will mark the -no phase as PENDING in TestStatus 
         if self._no_setup:
-            self._phases.remove(SETUP_PHASE)
+            i = self._phases.index(SETUP_PHASE)
```

Comment:
We leave the phase in but mark it so that it will be labeled as PEND in the TestStatus file